x/exchange/commitments.go

```diff
@@ -0,0 +1,306 @@
+func (c Commitment) Validate() error {
+	if _, err := sdk.AccAddressFromBech32(c.Account); err != nil {
+		return fmt.Errorf("invalid account %q: %w", c.Account, err)
+	}
+
+	if c.MarketId == 0 {
+		return errors.New("invalid market id: cannot be zero")
+	}
+
+	if err := c.Amount.Validate(); err != nil {
+		return fmt.Errorf("invalid amount %q: %w", c.Amount, err)
+	}
+
+	return nil
+}
+
+// String returns a string representation of this AccountAmount.
+func (a AccountAmount) String() string {
+	return fmt.Sprintf("%s:%q", a.Account, a.Amount)
+}
+
+// ValidateWithOptionalAmount returns an error if this AccountAmount is invalid. The amount is allowed to be empty.
+func (a AccountAmount) ValidateWithOptionalAmount() error {
+	if _, err := sdk.AccAddressFromBech32(a.Account); err != nil {
+		return fmt.Errorf("invalid account %q: %w", a.Account, err)
+	}
+	if err := a.Amount.Validate(); err != nil {
+		return fmt.Errorf("invalid amount %q: %w", a.Amount, err)
+	}
+	return nil
+}
+
+// Validate returns an error if this AccountAmount is invalid or has a zero amount.
+func (a AccountAmount) Validate() error {
+	if err := a.ValidateWithOptionalAmount(); err != nil {
+		return err
+	}
+	if a.Amount.IsZero() {
+		return fmt.Errorf("invalid amount %q: cannot be zero", a.Amount)
+	}
+	return nil
+}
+
+// SumAccountAmounts gets the total of all the amounts in the provided entries.
+func SumAccountAmounts(entries []AccountAmount) sdk.Coins {
+	var rv sdk.Coins
+	for _, entry := range entries {
+		rv = rv.Add(entry.Amount...)
+	}
+	return rv
+}
+
+// SimplifyAccountAmounts combines entries with the same account into a single entry.
+func SimplifyAccountAmounts(entries []AccountAmount) []AccountAmount {
+	if len(entries) <= 1 {
+		return entries
+	}
+
+	amounts := make(map[string]sdk.Coins)
+	addrs := make([]string, 0, len(entries))
+	for _, entry := range entries {
+		if _, known := amounts[entry.Account]; !known {
+			amounts[entry.Account] = nil
+			addrs = append(addrs, entry.Account)
+		}
+		amounts[entry.Account] = amounts[entry.Account].Add(entry.Amount...)
+	}
+
+	rv := make([]AccountAmount, len(addrs))
+	for i, addr := range addrs {
+		rv[i] = AccountAmount{Account: addr, Amount: amounts[addr]}
+	}
+
+	return rv
+}
+
+// AccountAmountsToBankInputs converts some AccountAmount entries, each to a banktypes.Input.
+func AccountAmountsToBankInputs(entries ...AccountAmount) []banktypes.Input {
+	rv := make([]banktypes.Input, len(entries))
+	for i, entry := range entries {
+		rv[i] = banktypes.Input{Address: entry.Account, Coins: entry.Amount}
+	}
+	return rv
+}
+
+// AccountAmountsToBankOutputs converts some AccountAmount entries, each to a banktypes.Output.
+func AccountAmountsToBankOutputs(entries ...AccountAmount) []banktypes.Output {
+	rv := make([]banktypes.Output, len(entries))
+	for i, entry := range entries {
+		rv[i] = banktypes.Output{Address: entry.Account, Coins: entry.Amount}
+	}
+	return rv
+}
+
```

Comment:
NIT: Can you rename the `some` to `each`?  This wording removes any ambiguity about how many entries are being converted.

Reply:
Fixed.